compiler/x/amd64/codegen/AMD64SystemLinkage.cpp

```diff
@@ -265,6 +265,79 @@ TR::AMD64Win64FastCallLinkage::AMD64Win64FastCallLinkage(TR::CodeGenerator *cg)
    TR_ASSERT(p == (machine()->getNumGlobalGPRs() + machine()->_numGlobalFPRs), "assertion failure");
    }
 
+// Build arguments for microsoft x64 calling convention dispatch.
+//
+int32_t TR::AMD64Win64FastCallLinkage::buildArgs(TR::Node* callNode, TR::RegisterDependencyConditions* deps)
+   {
+   static bool DuplicateParameterOnStack = feGetEnv("TR_DuplicateParameterOnStack");
+   static bool UseOldFastCallLinkage = feGetEnv("TR_UseOldFastCallLinkage");
+   if (UseOldFastCallLinkage || callNode->getOpCode().isIndirect())
+      {
+      return TR::AMD64SystemLinkage::buildArgs(callNode, deps);
```

Comment:
Is calling the parent implementation the correct thing to do if it already doesn't follow the parameter passing rules?

Reply:
Good question, I don't think the indirect route is ever excised. I don't see any OMR tests or OpenJ9 usage of this route. Should we raise an error if anyone gets into this route?